bitnami/keycloak/templates/statefulset.yaml

```diff
@@ -249,6 +249,11 @@ spec:
               containerPort: {{ .Values.containerPorts.https }}
               protocol: TCP
             {{- end }}
+            {{- if and (.Values.metrics.enabled) (not (eq (.Values.containerPorts.http | quote) (.Values.containerPorts.metrics | quote) )) }}
```

Comment:
When removing the 'quote' calling `helm template --set metrics.enabled=true --set containerPorts.metrics=8080 .` fails as well as calling `helm install` with the same parameters. (My helm Version: `version.BuildInfo{Version:"v3.16.2", GitCommit:"13654a52f7c70a143b1dd51416d633e1071faffb", GitTreeState:"", GoVersion:"go1.23.2"}`)


```
helm template --debug --set metrics.enabled=true --set containerPorts.metrics=8080 .                                                 1 ↵
install.go:224: 2024-11-04 20:27:38.309844737 +0100 CET m=+0.023248418 [debug] Original chart version: ""
install.go:241: 2024-11-04 20:27:38.309873812 +0100 CET m=+0.023277483 [debug] CHART PATH: /home/martin/src/bitnami_charts/bitnami/keycloak


Error: template: keycloak/templates/networkpolicy.yaml:69:51: executing "keycloak/templates/networkpolicy.yaml" at <eq (.Values.containerPorts.http) (.Values.containerPorts.metrics)>: error calling eq: incompatible types for comparison
helm.go:86: 2024-11-04 20:27:38.341824604 +0100 CET m=+0.055228285 [debug] template: keycloak/templates/networkpolicy.yaml:69:51: executing "keycloak/templates/networkpolicy.yaml" at <eq (.Values.containerPorts.http) (.Values.containerPorts.metrics)>: error calling eq: incompatible types for comparison
```
When using the Values file only the `quote` does not matter. Therefore, I did not remove the quote. However, I'm not a helm expert and may be doing things wrong here. Also, if this is intended behaviour I'm happily adding a commit removing the quote.

Reply:
I've just reproduce the issue you mentioned. `--set` parameter is parsing the number as float and the eq function can not compare an integer against a float. You can workaorund the issue with `--set-json` but I think is clever something like this:
```suggestion
            {{- if and (.Values.metrics.enabled) (not (eq (.Values.containerPorts.http | int) (.Values.containerPorts.metrics | int) )) }}
```

This will ensure that we have set integers in the ports